### ISSUE TITLE
[REVIEW]: ivadomed: A Medical Imaging Deep Learning Toolbox

### ISSUE THREAD

**Issue by whedon** [openjournals/joss-reviews#2868]
**Submitting author:** @charleygros (<a href="http://orcid.org/0000-0003-4318-0024">Charley Gros</a>)
**Repository:** <a href="https://github.com/ivadomed/ivadomed" target ="_blank">https://github.com/ivadomed/ivadomed</a>
**Version:** v2.7.1
**Editor:** @cMadan
**Reviewer:** @NMontanaBrown, @justusschock, @lbugnon
**Archive:** <a href="https://doi.org/10.5281/zenodo.4527802" target="_blank">10.5281/zenodo.4527802</a>

**:warning: JOSS reduced service mode :warning:**

Due to the challenges of the COVID-19 pandemic, JOSS is currently operating in a "reduced service mode". You can read more about what that means in [our blog post](https://blog.joss.theoj.org/2020/05/reopening-joss).

## Status

[![status](https://joss.theoj.org/papers/77101825a4bd5baa0ffaef17105f80b0/status.svg)](https://joss.theoj.org/papers/77101825a4bd5baa0ffaef17105f80b0)

Status badge code:

```
HTML: <a href="https://joss.theoj.org/papers/77101825a4bd5baa0ffaef17105f80b0"><img src="https://joss.theoj.org/papers/77101825a4bd5baa0ffaef17105f80b0/status.svg"></a>
Markdown: [![status](https://joss.theoj.org/papers/77101825a4bd5baa0ffaef17105f80b0/status.svg)](https://joss.theoj.org/papers/77101825a4bd5baa0ffaef17105f80b0)
```
**Reviewers and authors**:

Please avoid lengthy details of difficulties in the review thread. Instead, please create a new issue in the <a href="https://github.com/ivadomed/ivadomed" target="_blank">target repository</a> and link to those issues (especially acceptance-blockers) by leaving comments in the review thread below. (For completists: if the target issue tracker is also on GitHub, linking the review thread in the issue or vice versa will create corresponding breadcrumb trails in the link target.)

## Reviewer instructions & questions

@NMontanaBrown & @justusschock & @lbugnon, please carry out your review in this issue by updating the checklist below. If you cannot edit the checklist please:
1. Make sure you're logged in to your GitHub account
2. Be sure to accept the invite at this URL: https://github.com/openjournals/joss-reviews/invitations

The reviewer guidelines are available here: https://joss.readthedocs.io/en/latest/reviewer_guidelines.html. Any questions/concerns please let @cMadan know.

✨ **Please start on your review when you are able, and be sure to complete your review in the next six weeks, at the very latest** ✨ 

### Conflict of interest

- [x] I confirm that I have read the [JOSS conflict of interest (COI) policy](https://github.com/openjournals/joss/blob/master/COI.md) and that: I have no COIs with reviewing this work or that any perceived COIs have been waived by JOSS for the purpose of this review.

### Code of Conduct

- [x] I confirm that I read and will adhere to the [JOSS code of conduct](https://joss.theoj.org/about#code_of_conduct).

### General checks

- [x] **Repository:** Is the source code for this software available at the <a target="_blank" href="https://github.com/ivadomed/ivadomed">repository url</a>?
- [x] **License:** Does the repository contain a plain-text LICENSE file with the contents of an [OSI approved](https://opensource.org/licenses/alphabetical) software license?
- [x] **Contribution and authorship:** Has the submitting author (@charleygros) made major contributions to the software? Does the full list of paper authors seem appropriate and complete?
- [x] **Substantial scholarly effort:** Does this submission meet the scope eligibility described in the [JOSS guidelines](https://joss.readthedocs.io/en/latest/submitting.html#substantial-scholarly-effort)

### Functionality

- [x] **Installation:** Does installation proceed as outlined in the documentation?
- [x] **Functionality:** Have the functional claims of the software been confirmed?
- [x] **Performance:** If there are any performance claims of the software, have they been confirmed? (If there are no claims, please check off this item.)

### Documentation

- [x] **A statement of need:** Do the authors clearly state what problems the software is designed to solve and who the target audience is?
- [x] **Installation instructions:** Is there a clearly-stated list of dependencies? Ideally these should be handled with an automated package management solution.
- [x] **Example usage:** Do the authors include examples of how to use the software (ideally to solve real-world analysis problems).
- [x] **Functionality documentation:** Is the core functionality of the software documented to a satisfactory level (e.g., API method documentation)?
- [x] **Automated tests:** Are there automated tests or manual steps described so that the functionality of the software can be verified?
- [x] **Community guidelines:** Are there clear guidelines for third parties wishing to 1) Contribute to the software 2) Report issues or problems with the software 3) Seek support

### Software paper

- [x] **Summary:** Has a clear description of the high-level functionality and purpose of the software for a diverse, non-specialist audience been provided?
- [x] **A statement of need:** Do the authors clearly state what problems the software is designed to solve and who the target audience is?
- [x] **State of the field:** Do the authors describe how this software compares to other commonly-used packages?
- [x] **Quality of writing:** Is the paper well written (i.e., it does not require editing for structure, language, or writing quality)?
- [x] **References:** Is the list of references complete, and is everything cited appropriately that should be cited (e.g., papers, datasets, software)? Do references in the text use the proper [citation syntax]( https://rmarkdown.rstudio.com/authoring_bibliographies_and_citations.html#citation_syntax)?

### Conflict of interest

- [x] I confirm that I have read the [JOSS conflict of interest (COI) policy](https://github.com/openjournals/joss/blob/master/COI.md) and that: I have no COIs with reviewing this work or that any perceived COIs have been waived by JOSS for the purpose of this review.

### Code of Conduct

- [x] I confirm that I read and will adhere to the [JOSS code of conduct](https://joss.theoj.org/about#code_of_conduct).

### General checks

- [x] **Repository:** Is the source code for this software available at the <a target="_blank" href="https://github.com/ivadomed/ivadomed">repository url</a>?
- [x] **License:** Does the repository contain a plain-text LICENSE file with the contents of an [OSI approved](https://opensource.org/licenses/alphabetical) software license?
- [x] **Contribution and authorship:** Has the submitting author (@charleygros) made major contributions to the software? Does the full list of paper authors seem appropriate and complete?
- [x] **Substantial scholarly effort:** Does this submission meet the scope eligibility described in the [JOSS guidelines](https://joss.readthedocs.io/en/latest/submitting.html#substantial-scholarly-effort)

### Functionality

- [x] **Installation:** Does installation proceed as outlined in the documentation?
- [x] **Functionality:** Have the functional claims of the software been confirmed?
- [x] **Performance:** If there are any performance claims of the software, have they been confirmed? (If there are no claims, please check off this item.)

### Documentation

- [x] **A statement of need:** Do the authors clearly state what problems the software is designed to solve and who the target audience is?
- [x] **Installation instructions:** Is there a clearly-stated list of dependencies? Ideally these should be handled with an automated package management solution.
- [x] **Example usage:** Do the authors include examples of how to use the software (ideally to solve real-world analysis problems).
- [x] **Functionality documentation:** Is the core functionality of the software documented to a satisfactory level (e.g., API method documentation)?
- [x] **Automated tests:** Are there automated tests or manual steps described so that the functionality of the software can be verified?
- [x] **Community guidelines:** Are there clear guidelines for third parties wishing to 1) Contribute to the software 2) Report issues or problems with the software 3) Seek support

### Software paper

- [x] **Summary:** Has a clear description of the high-level functionality and purpose of the software for a diverse, non-specialist audience been provided?
- [x] **A statement of need:** Do the authors clearly state what problems the software is designed to solve and who the target audience is?
- [x] **State of the field:** Do the authors describe how this software compares to other commonly-used packages?
- [x] **Quality of writing:** Is the paper well written (i.e., it does not require editing for structure, language, or writing quality)?
- [x] **References:** Is the list of references complete, and is everything cited appropriately that should be cited (e.g., papers, datasets, software)? Do references in the text use the proper [citation syntax]( https://rmarkdown.rstudio.com/authoring_bibliographies_and_citations.html#citation_syntax)?

### Conflict of interest

- [x] I confirm that I have read the [JOSS conflict of interest (COI) policy](https://github.com/openjournals/joss/blob/master/COI.md) and that: I have no COIs with reviewing this work or that any perceived COIs have been waived by JOSS for the purpose of this review.

### Code of Conduct

- [x] I confirm that I read and will adhere to the [JOSS code of conduct](https://joss.theoj.org/about#code_of_conduct).

### General checks

- [x] **Repository:** Is the source code for this software available at the <a target="_blank" href="https://github.com/ivadomed/ivadomed">repository url</a>?
- [x] **License:** Does the repository contain a plain-text LICENSE file with the contents of an [OSI approved](https://opensource.org/licenses/alphabetical) software license?
- [x] **Contribution and authorship:** Has the submitting author (@charleygros) made major contributions to the software? Does the full list of paper authors seem appropriate and complete?
- [x] **Substantial scholarly effort:** Does this submission meet the scope eligibility described in the [JOSS guidelines](https://joss.readthedocs.io/en/latest/submitting.html#substantial-scholarly-effort)

### Functionality

- [x] **Installation:** Does installation proceed as outlined in the documentation?
- [x] **Functionality:** Have the functional claims of the software been confirmed?
- [x] **Performance:** If there are any performance claims of the software, have they been confirmed? (If there are no claims, please check off this item.)

### Documentation

- [x] **A statement of need:** Do the authors clearly state what problems the software is designed to solve and who the target audience is?
- [x] **Installation instructions:** Is there a clearly-stated list of dependencies? Ideally these should be handled with an automated package management solution.
- [x] **Example usage:** Do the authors include examples of how to use the software (ideally to solve real-world analysis problems).
- [x] **Functionality documentation:** Is the core functionality of the software documented to a satisfactory level (e.g., API method documentation)?
- [x] **Automated tests:** Are there automated tests or manual steps described so that the functionality of the software can be verified?
- [x] **Community guidelines:** Are there clear guidelines for third parties wishing to 1) Contribute to the software 2) Report issues or problems with the software 3) Seek support

### Software paper

- [x] **Summary:** Has a clear description of the high-level functionality and purpose of the software for a diverse, non-specialist audience been provided?
- [x] **A statement of need:** Do the authors clearly state what problems the software is designed to solve and who the target audience is?
- [x] **State of the field:** Do the authors describe how this software compares to other commonly-used packages?
- [x] **Quality of writing:** Is the paper well written (i.e., it does not require editing for structure, language, or writing quality)?
- [x] **References:** Is the list of references complete, and is everything cited appropriately that should be cited (e.g., papers, datasets, software)? Do references in the text use the proper [citation syntax]( https://rmarkdown.rstudio.com/authoring_bibliographies_and_citations.html#citation_syntax)?

**Comment by whedon** [comment]
Hello human, I'm @whedon, a robot that can help you with some common editorial tasks. @NMontanaBrown, @justusschock, @lbugnon it looks like you're currently assigned to review this paper :tada:.

**:warning: JOSS reduced service mode :warning:**

Due to the challenges of the COVID-19 pandemic, JOSS is currently operating in a "reduced service mode". You can read more about what that means in [our blog post](https://blog.joss.theoj.org/2020/05/reopening-joss).

**:star: Important :star:**

If you haven't already, you should seriously consider unsubscribing from GitHub notifications for this (https://github.com/openjournals/joss-reviews) repository. As a reviewer, you're probably currently watching this repository which means for GitHub's default behaviour you will receive notifications (emails) for all reviews 😿

To fix this do the following two things:

1. Set yourself as 'Not watching' https://github.com/openjournals/joss-reviews:

![watching](https://cloud.githubusercontent.com/assets/4483/20250593/64d7ce48-a9de-11e6-9225-d3dfb3e48e68.png)

2. You may also like to change your default settings for this watching repositories in your GitHub profile here: https://github.com/settings/notifications

![notifications](https://cloud.githubusercontent.com/assets/4483/20250604/79eb3144-a9de-11e6-9e51-0fe940966689.png)


For a list of things I can do to help you, just type:

```
@whedon commands
```

For example, to regenerate the paper pdf after making changes in the paper's md or bib files, type:

```
@whedon generate pdf
```

**Comment by whedon** [comment]
:point_right::page_facing_up: [Download article proof](https://raw.githubusercontent.com/openjournals/joss-papers/joss.02868/joss.02868/10.21105.joss.02868.pdf) :page_facing_up: [View article proof on GitHub](https://github.com/openjournals/joss-papers/blob/joss.02868/joss.02868/10.21105.joss.02868.pdf) :page_facing_up: :point_left:

**Comment by whedon** [comment]
```
Reference check summary (note 'MISSING' DOIs are suggestions that need verification):

OK DOIs

- 10.1109/iccv.2019.00707 is OK
- 10.1016/j.nicl.2018.03.029 is OK
- 10.1016/j.neuroimage.2018.09.081 is OK
- 10.14245/ns.1938396.198 is OK
- 10.1038/sdata.2016.44 is OK
- 10.1016/j.media.2019.101557 is OK
- 10.1016/j.neucom.2019.01.103 is OK
- 10.1007/978-3-319-46723-8_54 is OK
- 10.1109/iccvw.2017.9 is OK
- 10.1109/CVPR.2016.90 is OK
- 10.1016/j.neuroimage.2018.11.042 is OK
- 10.1109/3dv.2016.79 is OK
- 10.1007/978-3-030-00931-1_70 is OK

MISSING DOIs

- None

INVALID DOIs

- None
```

**Comment by whedon** [comment]
:wave: @justusschock, please update us on how your review is going.

**Comment by whedon** [comment]
:wave: @NMontanaBrown, please update us on how your review is going.

**Comment by whedon** [comment]
:wave: @lbugnon, please update us on how your review is going.

**Comment by justusschock** [comment]
* __Installation:__ With Python 3.9 it does not work, although the documentation states, it should just be Python>=3.6
```bash
pip install ivadomed
Collecting ivadomed
  Downloading ivadomed-2.6.0-py3-none-any.whl (139 kB)
     |████████████████████████████████| 139 kB 8.9 MB/s
  Downloading ivadomed-2.5.0-py3-none-any.whl (137 kB)
     |████████████████████████████████| 137 kB 17.9 MB/s
  Downloading ivadomed-2.4.0-py3-none-any.whl (132 kB)
     |████████████████████████████████| 132 kB 23.7 MB/s
  Downloading ivadomed-2.3.1-py3-none-any.whl (129 kB)
     |████████████████████████████████| 129 kB 24.1 MB/s
  Downloading ivadomed-2.2.1-py3-none-any.whl (127 kB)
     |████████████████████████████████| 127 kB 28.2 MB/s
  Downloading ivadomed-2.2.0-py3-none-any.whl (127 kB)
     |████████████████████████████████| 127 kB 23.9 MB/s
  Downloading ivadomed-2.1.0-py3-none-any.whl (110 kB)
     |████████████████████████████████| 110 kB 30.8 MB/s
  Downloading ivadomed-2.0.2-py3-none-any.whl (108 kB)
     |████████████████████████████████| 108 kB 29.9 MB/s
  Downloading ivadomed-2.0.1-py3-none-any.whl (108 kB)
     |████████████████████████████████| 108 kB 30.9 MB/s
  Downloading ivadomed-1.2.1-py3-none-any.whl (68 kB)
     |████████████████████████████████| 68 kB 1.8 MB/s
  Downloading ivadomed-1.2-py3-none-any.whl (68 kB)
     |████████████████████████████████| 68 kB 5.0 MB/s
  Downloading ivadomed-1.1-py3-none-any.whl (68 kB)
     |████████████████████████████████| 68 kB 4.6 MB/s
  Downloading ivadomed-1.0-py3-none-any.whl (68 kB)
     |████████████████████████████████| 68 kB 2.4 MB/s
  Downloading ivadomed-0.14-py3-none-any.whl (68 kB)
     |████████████████████████████████| 68 kB 4.8 MB/s
  Downloading ivadomed-0.13-py3-none-any.whl (68 kB)
     |████████████████████████████████| 68 kB 457 kB/s
  Downloading ivadomed-0.12-py3-none-any.whl (66 kB)
     |████████████████████████████████| 66 kB 3.5 MB/s
  Downloading ivadomed-0.11-py3-none-any.whl (66 kB)
     |████████████████████████████████| 66 kB 3.3 MB/s
ERROR: Cannot install ivadomed==0.11, ivadomed==0.12, ivadomed==0.13, ivadomed==0.14, ivadomed==1.0, ivadomed==1.1, ivadomed==1.2, ivadomed==1.2.1, ivadomed==2.0.1, ivadomed==2.0.2, ivadomed==2.1.0, ivadomed==2.2.0, ivadomed==2.2.1, ivadomed==2.3.1, ivadomed==2.4.0, ivadomed==2.5.0 and ivadomed==2.6.0 because these package versions have conflicting dependencies.

The conflict is caused by:
    ivadomed 2.6.0 depends on onnxruntime==1.4.0
    ivadomed 2.5.0 depends on onnxruntime==1.4.0
    ivadomed 2.4.0 depends on onnxruntime
    ivadomed 2.3.1 depends on torchvision>=0.6.0
    ivadomed 2.2.1 depends on torchvision>=0.6.0
    ivadomed 2.2.0 depends on torchvision>=0.6.0
    ivadomed 2.1.0 depends on torchvision>=0.6.0
    ivadomed 2.0.2 depends on torchvision>=0.6.0
    ivadomed 2.0.1 depends on torchvision>=0.6.0
    ivadomed 1.2.1 depends on torchvision>=0.6.0
    ivadomed 1.2 depends on torchvision>=0.6.0
    ivadomed 1.1 depends on torchvision>=0.6.0
    ivadomed 1.0 depends on torchvision>=0.6.0
    ivadomed 0.14 depends on torchvision>=0.6.0
    ivadomed 0.13 depends on torchvision>=0.6.0
    ivadomed 0.12 depends on torchvision>=0.6.0
    ivadomed 0.11 depends on torchvision>=0.6.0

To fix this you could try to:
1. loosen the range of package versions you've specified
2. remove package versions to allow pip attempt to solve the dependency conflict

ERROR: ResolutionImpossible: for help visit https://pip.pypa.io/en/latest/user_guide/#fixing-conflicting-dependencies
```
With Python 3.8.5 it works fine! Maybe having the installation instructions also in the readme and not only in the linked documentation would be nice.

* __State of Need:__
While it is clearly stated, why the software is necessary and how it compares to other available software, the table at https://ivadomed.org/en/latest/purpose.html#comparison-with-other-projects is not nice to look at. Maybe divide it into several tables with major parts.

* __Automated tests__: While tests are present and coverage is reported correctly, it would be nice having a badge pointing to the actual outcome of GitHub actions.

* __Community Guidelines__: While they are available at https://ivadomed.org/en/latest/contributing.html and in the repository wiki, it is not the usual place for GitHub repositories. SO it would be nice to also have a file like `CODE_OF_CONDUCT.md` (which must only point to the available resources) as well as issue and Pull-Request templates pointing to the relevant parts of the wiki.

**Comment by charleygros** [comment]
Many thanks for taking the time to review our work and providing valuable feedback @justusschock !
 
> * **Installation:** With Python 3.9 it does not work, although the documentation states, it should just be Python>=3.6

Thank you for raising this point! We modified our documentation accordingly, see [PR #562](https://github.com/ivadomed/ivadomed/pull/562), stating that ivadomed works for Python >= 3.6 and < 3.9.

> With Python 3.8.5 it works fine! Maybe having the installation instructions also in the readme and not only in the linked documentation would be nice.

We agree, thanks for suggesting this. We added the installation instructions in the README as well, see [PR #561](https://github.com/ivadomed/ivadomed/pull/561).

>   While it is clearly stated, why the software is necessary and how it compares to other available software, the table at https://ivadomed.org/en/latest/purpose.html#comparison-with-other-projects is not nice to look at. Maybe divide it into several tables with major parts.

We modified the layout of the table by transposing it and making them more compact, see [PR #567](https://github.com/ivadomed/ivadomed/pull/567).

> While tests are present and coverage is reported correctly, it would be nice having a badge pointing to the actual outcome of GitHub actions.

We added badges pointing to the Github action panels in [PR #566](https://github.com/ivadomed/ivadomed/pull/566).

> While they are available at https://ivadomed.org/en/latest/contributing.html and in the repository wiki, it is not the usual place for GitHub repositories. SO it would be nice to also have a file like `CODE_OF_CONDUCT.md` (which must only point to the available resources) as well as issue and Pull-Request templates pointing to the relevant parts of the wiki.

We agree with you @justusschock, thank you for raising this point. In the [PR #565](https://github.com/ivadomed/ivadomed/pull/565), we did the following changes:
- Add 3 templates (1 for PRs and 2 for issues)
- Add a CODE_OF_CONDUCT.md that uses the Contributor Covenant
- Move the contributing.rst to the project root to make it more visible, and add a symlink within the docs, so that it can appear in both places at once.

**Comment by lbugnon** [comment]
Hi @charleygros, just a minor thing,  `virtualenv venv-ivadomed --python=python3.6` gives an error (because i have installed python 3.7, I think It could be omited or stated to look into the OS.

**Comment by lbugnon** [comment]
Usage (as a demo) is not clear, for instance `ivadomed ivadomed/config/config.json` doesn't work. Adding -c option lead to error RuntimeError: The dataset_description.json file was not found.

**Comment by lbugnon** [comment]
@charleygros referneces are missing in the paper md

**Comment by charleygros** [comment]
Hi @lbugnon , We really appreciate that you took time to review our work!

> virtualenv venv-ivadomed --python=python3.6 gives an error (because i have installed python 3.7, I think It could be omited or stated to look into the OS

Thanks for raising this issue: it has been fixed in[ PR #579](https://github.com/ivadomed/ivadomed/pull/579) (omission of the Python version).

> Usage (as a demo) is not clear, for instance ivadomed ivadomed/config/config.json doesn't work. Adding -c option lead to error RuntimeError: The dataset_description.json file was not found. I think a "Please first look into the tutorials" would work

We agree with you @lbugnon: we added `-c` and directed the reader towards the tutorials for additional info, see [PR #578](https://github.com/ivadomed/ivadomed/pull/578).

> referneces are missing in the paper md

References are in `paper.bib` and correctly appear in the compiled document, see [here](https://raw.githubusercontent.com/openjournals/joss-papers/joss.02868/joss.02868/10.21105.joss.02868.pdf) or via [Wheldom paper preview](Whedon paper preview service). Please let me know if I am missing something :-)

**Comment by lbugnon** [comment]
@charleygros ok, I think it's a great work!

**Comment by NMontanaBrown** [comment]
> Hey @charleygros,
> 
> I really like your software. However, I have some small comments:
> 
> * **Installation:** With Python 3.9 it does not work, although the documentation states, it should just be Python>=3.6
> 
> ```shell
> pip install ivadomed
> Collecting ivadomed
>   Downloading ivadomed-2.6.0-py3-none-any.whl (139 kB)
>      |████████████████████████████████| 139 kB 8.9 MB/s
>   Downloading ivadomed-2.5.0-py3-none-any.whl (137 kB)
>      |████████████████████████████████| 137 kB 17.9 MB/s
>   Downloading ivadomed-2.4.0-py3-none-any.whl (132 kB)
>      |████████████████████████████████| 132 kB 23.7 MB/s
>   Downloading ivadomed-2.3.1-py3-none-any.whl (129 kB)
>      |████████████████████████████████| 129 kB 24.1 MB/s
>   Downloading ivadomed-2.2.1-py3-none-any.whl (127 kB)
>      |████████████████████████████████| 127 kB 28.2 MB/s
>   Downloading ivadomed-2.2.0-py3-none-any.whl (127 kB)
>      |████████████████████████████████| 127 kB 23.9 MB/s
>   Downloading ivadomed-2.1.0-py3-none-any.whl (110 kB)
>      |████████████████████████████████| 110 kB 30.8 MB/s
>   Downloading ivadomed-2.0.2-py3-none-any.whl (108 kB)
>      |████████████████████████████████| 108 kB 29.9 MB/s
>   Downloading ivadomed-2.0.1-py3-none-any.whl (108 kB)
>      |████████████████████████████████| 108 kB 30.9 MB/s
>   Downloading ivadomed-1.2.1-py3-none-any.whl (68 kB)
>      |████████████████████████████████| 68 kB 1.8 MB/s
>   Downloading ivadomed-1.2-py3-none-any.whl (68 kB)
>      |████████████████████████████████| 68 kB 5.0 MB/s
>   Downloading ivadomed-1.1-py3-none-any.whl (68 kB)
>      |████████████████████████████████| 68 kB 4.6 MB/s
>   Downloading ivadomed-1.0-py3-none-any.whl (68 kB)
>      |████████████████████████████████| 68 kB 2.4 MB/s
>   Downloading ivadomed-0.14-py3-none-any.whl (68 kB)
>      |████████████████████████████████| 68 kB 4.8 MB/s
>   Downloading ivadomed-0.13-py3-none-any.whl (68 kB)
>      |████████████████████████████████| 68 kB 457 kB/s
>   Downloading ivadomed-0.12-py3-none-any.whl (66 kB)
>      |████████████████████████████████| 66 kB 3.5 MB/s
>   Downloading ivadomed-0.11-py3-none-any.whl (66 kB)
>      |████████████████████████████████| 66 kB 3.3 MB/s
> ERROR: Cannot install ivadomed==0.11, ivadomed==0.12, ivadomed==0.13, ivadomed==0.14, ivadomed==1.0, ivadomed==1.1, ivadomed==1.2, ivadomed==1.2.1, ivadomed==2.0.1, ivadomed==2.0.2, ivadomed==2.1.0, ivadomed==2.2.0, ivadomed==2.2.1, ivadomed==2.3.1, ivadomed==2.4.0, ivadomed==2.5.0 and ivadomed==2.6.0 because these package versions have conflicting dependencies.
> 
> The conflict is caused by:
>     ivadomed 2.6.0 depends on onnxruntime==1.4.0
>     ivadomed 2.5.0 depends on onnxruntime==1.4.0
>     ivadomed 2.4.0 depends on onnxruntime
>     ivadomed 2.3.1 depends on torchvision>=0.6.0
>     ivadomed 2.2.1 depends on torchvision>=0.6.0
>     ivadomed 2.2.0 depends on torchvision>=0.6.0
>     ivadomed 2.1.0 depends on torchvision>=0.6.0
>     ivadomed 2.0.2 depends on torchvision>=0.6.0
>     ivadomed 2.0.1 depends on torchvision>=0.6.0
>     ivadomed 1.2.1 depends on torchvision>=0.6.0
>     ivadomed 1.2 depends on torchvision>=0.6.0
>     ivadomed 1.1 depends on torchvision>=0.6.0
>     ivadomed 1.0 depends on torchvision>=0.6.0
>     ivadomed 0.14 depends on torchvision>=0.6.0
>     ivadomed 0.13 depends on torchvision>=0.6.0
>     ivadomed 0.12 depends on torchvision>=0.6.0
>     ivadomed 0.11 depends on torchvision>=0.6.0
> 
> To fix this you could try to:
> 1. loosen the range of package versions you've specified
> 2. remove package versions to allow pip attempt to solve the dependency conflict
> 
> ERROR: ResolutionImpossible: for help visit https://pip.pypa.io/en/latest/user_guide/#fixing-conflicting-dependencies
> ```
> 
> With Python 3.8.5 it works fine! Maybe having the installation instructions also in the readme and not only in the linked documentation would be nice.
> 
> * **State of Need:**
>   While it is clearly stated, why the software is necessary and how it compares to other available software, the table at https://ivadomed.org/en/latest/purpose.html#comparison-with-other-projects is not nice to look at. Maybe divide it into several tables with major parts.
> * **Automated tests**: While tests are present and coverage is reported correctly, it would be nice having a badge pointing to the actual outcome of GitHub actions.
> * **Community Guidelines**: While they are available at https://ivadomed.org/en/latest/contributing.html and in the repository wiki, it is not the usual place for GitHub repositories. SO it would be nice to also have a file like `CODE_OF_CONDUCT.md` (which must only point to the available resources) as well as issue and Pull-Request templates pointing to the relevant parts of the wiki.

Hi 👋 @charleygros, thanks for the work.

* Installation
Currently I am having the same problems with installation as @lbugnon reported earlier, with differing outcomes depending on OS.
MacOS installation proceeds correctly for latest release.
Ubuntu 18.04.3 installation proceeds correctly for latest release.
Windows OS left me with the same error as above, for some reason.

` pip install ivadomed
Collecting ivadomed
  Using cached ivadomed-2.6.1-py3-none-any.whl (145 kB)
  Using cached ivadomed-2.6.0-py3-none-any.whl (139 kB)
  Using cached ivadomed-2.5.0-py3-none-any.whl (137 kB)
  Using cached ivadomed-2.4.0-py3-none-any.whl (132 kB)
  Using cached ivadomed-2.3.1-py3-none-any.whl (129 kB)
  Using cached ivadomed-2.2.1-py3-none-any.whl (127 kB)
  Using cached ivadomed-2.2.0-py3-none-any.whl (127 kB)
  Using cached ivadomed-2.1.0-py3-none-any.whl (110 kB)
  Using cached ivadomed-2.0.2-py3-none-any.whl (108 kB)
  Using cached ivadomed-2.0.1-py3-none-any.whl (108 kB)
  Using cached ivadomed-1.2.1-py3-none-any.whl (68 kB)
  Using cached ivadomed-1.2-py3-none-any.whl (68 kB)
  Using cached ivadomed-1.1-py3-none-any.whl (68 kB)
  Using cached ivadomed-1.0-py3-none-any.whl (68 kB)
  Using cached ivadomed-0.14-py3-none-any.whl (68 kB)
  Using cached ivadomed-0.13-py3-none-any.whl (68 kB)
  Using cached ivadomed-0.12-py3-none-any.whl (66 kB)
  Using cached ivadomed-0.11-py3-none-any.whl (66 kB)
ERROR: Cannot install ivadomed==0.11, ivadomed==0.12, ivadomed==0.13, ivadomed==0.14, ivadomed==1.0, ivadomed==1.1, ivadomed==1.2, ivadomed==1.2.1, ivadomed==2.0.1, ivadomed==2.0.2, ivadomed==2.1.0, ivadomed==2.2.0, ivadomed==2.2.1, ivadomed==2.3.1, ivadomed==2.4.0, ivadomed==2.5.0, ivadomed==2.6.0 and ivadomed==2.6.1 because these package versions have conflicting dependencies.

The conflict is caused by:
    ivadomed 2.6.1 depends on torchvision==0.6.0
    ivadomed 2.6.0 depends on torchvision==0.6.0
    ivadomed 2.5.0 depends on torchvision==0.6.0
    ivadomed 2.4.0 depends on torchvision==0.6.0
    ivadomed 2.3.1 depends on torchvision>=0.6.0
    ivadomed 2.2.1 depends on torchvision>=0.6.0
    ivadomed 2.2.0 depends on torchvision>=0.6.0
    ivadomed 2.1.0 depends on torchvision>=0.6.0
    ivadomed 2.0.2 depends on torchvision>=0.6.0
    ivadomed 2.0.1 depends on torchvision>=0.6.0
    ivadomed 1.2.1 depends on torchvision>=0.6.0
    ivadomed 1.2 depends on torchvision>=0.6.0
    ivadomed 1.1 depends on torchvision>=0.6.0
    ivadomed 1.0 depends on torchvision>=0.6.0
    ivadomed 0.14 depends on torchvision>=0.6.0
    ivadomed 0.13 depends on torchvision>=0.6.0
    ivadomed 0.12 depends on torchvision>=0.6.0
    ivadomed 0.11 depends on torchvision>=0.6.0

To fix this you could try to:
1. loosen the range of package versions you've specified
2. remove package versions to allow pip attempt to solve the dependency conflict

ERROR: ResolutionImpossible: for help visit https://pip.pypa.io/en/latest/user_guide/#fixing-conflicting-dependencies
`
Specifically, the error seems to be with torchvision:

` pip install ivadomed==2.6.1
Collecting ivadomed==2.6.1
  Using cached ivadomed-2.6.1-py3-none-any.whl (145 kB)
ERROR: Could not find a version that satisfies the requirement torchvision==0.6.0 (from ivadomed)
ERROR: No matching distribution found for torchvision==0.6.0
`

I tried this with various python versions to no avail. Overall, this seems to be related to WindowsOS, as the installation proceeds fine on other OS.  I would suggest updating your RTD installation section and the README to state installation works as described for MacOS and Linux, but not currently supported for Windows. It may be something you want to fix or not - but this will clarify upfront to users that installation works as intended except on Windows and avoid head scratching at the first stage.

**Comment by jcohenadad** [comment]
The [following reference](https://github.com/ivadomed/ivadomed/blob/master/paper.bib#L47), which is supposed to reference Pytorch, whould probably be replaced by this one:
````
@incollection{NEURIPS2019_9015,
title = {PyTorch: An Imperative Style, High-Performance Deep Learning Library},
author = {Paszke, Adam and Gross, Sam and Massa, Francisco and Lerer, Adam and Bradbury, James and Chanan, Gregory and Killeen, Trevor and Lin, Zeming and Gimelshein, Natalia and Antiga, Luca and Desmaison, Alban and Kopf, Andreas and Yang, Edward and DeVito, Zachary and Raison, Martin and Tejani, Alykhan and Chilamkurthy, Sasank and Steiner, Benoit and Fang, Lu and Bai, Junjie and Chintala, Soumith},
booktitle = {Advances in Neural Information Processing Systems 32},
editor = {H. Wallach and H. Larochelle and A. Beygelzimer and F. d\textquotesingle Alch\'{e}-Buc and E. Fox and R. Garnett},
pages = {8024--8035},
year = {2019},
publisher = {Curran Associates, Inc.},
url = {http://papers.neurips.cc/paper/9015-pytorch-an-imperative-style-high-performance-deep-learning-library.pdf}
}

source: https://github.com/pytorch/pytorch/blob/master/CITATION
````

**Comment by charleygros** [comment]
> The [following reference](https://github.com/ivadomed/ivadomed/blob/master/paper.bib#L47), which is supposed to reference Pytorch, whould probably be replaced by this one

Thanks for your suggestion @jcohenadad. Addressed in this [Pull Request](https://github.com/ivadomed/ivadomed/pull/616).

**Comment by charleygros** [comment]
Many thanks for taking the time to review our work and providing valuable feedback @NMontanaBrown!

> I tried this with various python versions to no avail. Overall, this seems to be related to WindowsOS, as the installation proceeds fine on other OS. I would suggest updating your RTD installation section and the README to state installation works as described for MacOS and Linux, but not currently supported for Windows. It may be something you want to fix or not - but this will clarify upfront to users that installation works as intended except on Windows and avoid head scratching at the first stage.

Thank you for raising this point! We followed your advice by:
- Opening a Feature Request to add compatibility to Windows → https://github.com/ivadomed/ivadomed/issues/604 
- Adding  explanation for Windows incompatibility in installation docs → https://github.com/ivadomed/ivadomed/pull/605 
- Specify Python version in setup.py → https://github.com/ivadomed/ivadomed/pull/603 

> In your table you include some examples of registration packages that use deep learning, yet ivadomed does not yet cover these functionalities (yet (?)). Are you planning on covering this use case in the future? Additionally, would you consider explaining this in your paper, and adding [DeepReg](https://doi.org/10.21105/joss.02705) to your table/citing it?

Thank you for this relevant suggestion! Indeed, `ivadomed` does not cover registration (yet). We have added a column for `DeepReg`, see [here](https://docs.google.com/spreadsheets/d/1_MydnHnlOAuYzJ9QBCvPC9Jq2xUmPWI-XttTfcdtW2Y/edit?usp=sharing) (Note: it will appear [here](https://ivadomed.org/en/latest/purpose.html) after the release). Please feel free to comment/suggest modifications if we missed some features of `DeepReg`.

**Comment by NMontanaBrown** [comment]
> Many thanks for taking the time to review our work and providing valuable feedback @NMontanaBrown!
> 
> > I tried this with various python versions to no avail. Overall, this seems to be related to WindowsOS, as the installation proceeds fine on other OS. I would suggest updating your RTD installation section and the README to state installation works as described for MacOS and Linux, but not currently supported for Windows. It may be something you want to fix or not - but this will clarify upfront to users that installation works as intended except on Windows and avoid head scratching at the first stage.
> 
> Thank you for raising this point! We followed your advice by:
> 
> * Opening a Feature Request to add compatibility to Windows → [ivadomed/ivadomed#604](https://github.com/ivadomed/ivadomed/issues/604)
> * Adding  explanation for Windows incompatibility in installation docs → [ivadomed/ivadomed#605](https://github.com/ivadomed/ivadomed/pull/605)
> * Specify Python version in setup.py → [ivadomed/ivadomed#603](https://github.com/ivadomed/ivadomed/pull/603)
> 
> > In your table you include some examples of registration packages that use deep learning, yet ivadomed does not yet cover these functionalities (yet (?)). Are you planning on covering this use case in the future? Additionally, would you consider explaining this in your paper, and adding [DeepReg](https://doi.org/10.21105/joss.02705) to your table/citing it?
> 
> Thank you for this relevant suggestion! Indeed, `ivadomed` does not cover registration (yet). We have added a column for `DeepReg`, see [PR618](https://github.com/ivadomed/ivadomed/pull/618). Please feel free to comment/suggest modifications if we missed some features of `DeepReg` (see preview [here](https://ivado-medical-imaging--618.org.readthedocs.build/en/618/purpose.html)).

Hi @charleygros, thanks for the response, the reference for DeepReg is accurate in your able. Now just waiting on resolution of [the docs update re Windows](https://github.com/ivadomed/ivadomed/pull/605), then all good to go from me 😄

**Comment by charleygros** [comment]
> Now just waiting on resolution of [the docs update re Windows](https://github.com/ivadomed/ivadomed/pull/605), then all good to go from me smile

Now merged, as well as an [additional PR](https://github.com/ivadomed/ivadomed/pull/621) about the same matter. Thanks again @NMontanaBrown :-)

**Comment by charleygros** [comment]
Hi @lbugnon,

I noticed you have not checked off the "Performance" item:

> Performance: If there are any performance claims of the software, have they been confirmed? (If there are no claims, please check off this item.)

May I ask if you have some feedback for us? Cheers

**Comment by lbugnon** [comment]
> Hi @lbugnon,
> 
> I noticed you have not checked off the "Performance" item:
> 
> > Performance: If there are any performance claims of the software, have they been confirmed? (If there are no claims, please check off this item.)
> 
> May I ask if you have some feedback for us? Cheers

Sorry, all fine by me!

**Comment by charleygros** [comment]
60 / 60 :white_check_mark:
Thanks again @NMontanaBrown, @lbugnon and @justusschock for your precious feedback!

@cMadan: What would be the next steps?

**Comment by ssinad** [comment]
> > Hey @charleygros,
> > I really like your software. However, I have some small comments:
> > 
> > * **Installation:** With Python 3.9 it does not work, although the documentation states, it should just be Python>=3.6
> > 
> > ```shell
> > pip install ivadomed
> > Collecting ivadomed
> >   Downloading ivadomed-2.6.0-py3-none-any.whl (139 kB)
> >      |████████████████████████████████| 139 kB 8.9 MB/s
> >   Downloading ivadomed-2.5.0-py3-none-any.whl (137 kB)
> >      |████████████████████████████████| 137 kB 17.9 MB/s
> >   Downloading ivadomed-2.4.0-py3-none-any.whl (132 kB)
> >      |████████████████████████████████| 132 kB 23.7 MB/s
> >   Downloading ivadomed-2.3.1-py3-none-any.whl (129 kB)
> >      |████████████████████████████████| 129 kB 24.1 MB/s
> >   Downloading ivadomed-2.2.1-py3-none-any.whl (127 kB)
> >      |████████████████████████████████| 127 kB 28.2 MB/s
> >   Downloading ivadomed-2.2.0-py3-none-any.whl (127 kB)
> >      |████████████████████████████████| 127 kB 23.9 MB/s
> >   Downloading ivadomed-2.1.0-py3-none-any.whl (110 kB)
> >      |████████████████████████████████| 110 kB 30.8 MB/s
> >   Downloading ivadomed-2.0.2-py3-none-any.whl (108 kB)
> >      |████████████████████████████████| 108 kB 29.9 MB/s
> >   Downloading ivadomed-2.0.1-py3-none-any.whl (108 kB)
> >      |████████████████████████████████| 108 kB 30.9 MB/s
> >   Downloading ivadomed-1.2.1-py3-none-any.whl (68 kB)
> >      |████████████████████████████████| 68 kB 1.8 MB/s
> >   Downloading ivadomed-1.2-py3-none-any.whl (68 kB)
> >      |████████████████████████████████| 68 kB 5.0 MB/s
> >   Downloading ivadomed-1.1-py3-none-any.whl (68 kB)
> >      |████████████████████████████████| 68 kB 4.6 MB/s
> >   Downloading ivadomed-1.0-py3-none-any.whl (68 kB)
> >      |████████████████████████████████| 68 kB 2.4 MB/s
> >   Downloading ivadomed-0.14-py3-none-any.whl (68 kB)
> >      |████████████████████████████████| 68 kB 4.8 MB/s
> >   Downloading ivadomed-0.13-py3-none-any.whl (68 kB)
> >      |████████████████████████████████| 68 kB 457 kB/s
> >   Downloading ivadomed-0.12-py3-none-any.whl (66 kB)
> >      |████████████████████████████████| 66 kB 3.5 MB/s
> >   Downloading ivadomed-0.11-py3-none-any.whl (66 kB)
> >      |████████████████████████████████| 66 kB 3.3 MB/s
> > ERROR: Cannot install ivadomed==0.11, ivadomed==0.12, ivadomed==0.13, ivadomed==0.14, ivadomed==1.0, ivadomed==1.1, ivadomed==1.2, ivadomed==1.2.1, ivadomed==2.0.1, ivadomed==2.0.2, ivadomed==2.1.0, ivadomed==2.2.0, ivadomed==2.2.1, ivadomed==2.3.1, ivadomed==2.4.0, ivadomed==2.5.0 and ivadomed==2.6.0 because these package versions have conflicting dependencies.
> > 
> > The conflict is caused by:
> >     ivadomed 2.6.0 depends on onnxruntime==1.4.0
> >     ivadomed 2.5.0 depends on onnxruntime==1.4.0
> >     ivadomed 2.4.0 depends on onnxruntime
> >     ivadomed 2.3.1 depends on torchvision>=0.6.0
> >     ivadomed 2.2.1 depends on torchvision>=0.6.0
> >     ivadomed 2.2.0 depends on torchvision>=0.6.0
> >     ivadomed 2.1.0 depends on torchvision>=0.6.0
> >     ivadomed 2.0.2 depends on torchvision>=0.6.0
> >     ivadomed 2.0.1 depends on torchvision>=0.6.0
> >     ivadomed 1.2.1 depends on torchvision>=0.6.0
> >     ivadomed 1.2 depends on torchvision>=0.6.0
> >     ivadomed 1.1 depends on torchvision>=0.6.0
> >     ivadomed 1.0 depends on torchvision>=0.6.0
> >     ivadomed 0.14 depends on torchvision>=0.6.0
> >     ivadomed 0.13 depends on torchvision>=0.6.0
> >     ivadomed 0.12 depends on torchvision>=0.6.0
> >     ivadomed 0.11 depends on torchvision>=0.6.0
> > 
> > To fix this you could try to:
> > 1. loosen the range of package versions you've specified
> > 2. remove package versions to allow pip attempt to solve the dependency conflict
> > 
> > ERROR: ResolutionImpossible: for help visit https://pip.pypa.io/en/latest/user_guide/#fixing-conflicting-dependencies
> > ```
> > 
> > 
> > With Python 3.8.5 it works fine! Maybe having the installation instructions also in the readme and not only in the linked documentation would be nice.
> > 
> > * **State of Need:**
> >   While it is clearly stated, why the software is necessary and how it compares to other available software, the table at https://ivadomed.org/en/latest/purpose.html#comparison-with-other-projects is not nice to look at. Maybe divide it into several tables with major parts.
> > * **Automated tests**: While tests are present and coverage is reported correctly, it would be nice having a badge pointing to the actual outcome of GitHub actions.
> > * **Community Guidelines**: While they are available at https://ivadomed.org/en/latest/contributing.html and in the repository wiki, it is not the usual place for GitHub repositories. SO it would be nice to also have a file like `CODE_OF_CONDUCT.md` (which must only point to the available resources) as well as issue and Pull-Request templates pointing to the relevant parts of the wiki.
> 
> Hi 👋 @charleygros, thanks for the work.
> 
> # Installation
> Currently I am having the same problems with installation as @lbugnon reported earlier, with differing outcomes depending on OS.
> MacOS installation proceeds correctly for latest release.
> Ubuntu 18.04.3 installation proceeds correctly for latest release.
> Windows OS left me with the same error as above, for some reason.
> 
> ```
> Collecting ivadomed
>   Using cached ivadomed-2.6.1-py3-none-any.whl (145 kB)
>   Using cached ivadomed-2.6.0-py3-none-any.whl (139 kB)
>   Using cached ivadomed-2.5.0-py3-none-any.whl (137 kB)
>   Using cached ivadomed-2.4.0-py3-none-any.whl (132 kB)
>   Using cached ivadomed-2.3.1-py3-none-any.whl (129 kB)
>   Using cached ivadomed-2.2.1-py3-none-any.whl (127 kB)
>   Using cached ivadomed-2.2.0-py3-none-any.whl (127 kB)
>   Using cached ivadomed-2.1.0-py3-none-any.whl (110 kB)
>   Using cached ivadomed-2.0.2-py3-none-any.whl (108 kB)
>   Using cached ivadomed-2.0.1-py3-none-any.whl (108 kB)
>   Using cached ivadomed-1.2.1-py3-none-any.whl (68 kB)
>   Using cached ivadomed-1.2-py3-none-any.whl (68 kB)
>   Using cached ivadomed-1.1-py3-none-any.whl (68 kB)
>   Using cached ivadomed-1.0-py3-none-any.whl (68 kB)
>   Using cached ivadomed-0.14-py3-none-any.whl (68 kB)
>   Using cached ivadomed-0.13-py3-none-any.whl (68 kB)
>   Using cached ivadomed-0.12-py3-none-any.whl (66 kB)
>   Using cached ivadomed-0.11-py3-none-any.whl (66 kB)
> ERROR: Cannot install ivadomed==0.11, ivadomed==0.12, ivadomed==0.13, ivadomed==0.14, ivadomed==1.0, ivadomed==1.1, ivadomed==1.2, ivadomed==1.2.1, ivadomed==2.0.1, ivadomed==2.0.2, ivadomed==2.1.0, ivadomed==2.2.0, ivadomed==2.2.1, ivadomed==2.3.1, ivadomed==2.4.0, ivadomed==2.5.0, ivadomed==2.6.0 and ivadomed==2.6.1 because these package versions have conflicting dependencies.
> 
> The conflict is caused by:
>     ivadomed 2.6.1 depends on torchvision==0.6.0
>     ivadomed 2.6.0 depends on torchvision==0.6.0
>     ivadomed 2.5.0 depends on torchvision==0.6.0
>     ivadomed 2.4.0 depends on torchvision==0.6.0
>     ivadomed 2.3.1 depends on torchvision>=0.6.0
>     ivadomed 2.2.1 depends on torchvision>=0.6.0
>     ivadomed 2.2.0 depends on torchvision>=0.6.0
>     ivadomed 2.1.0 depends on torchvision>=0.6.0
>     ivadomed 2.0.2 depends on torchvision>=0.6.0
>     ivadomed 2.0.1 depends on torchvision>=0.6.0
>     ivadomed 1.2.1 depends on torchvision>=0.6.0
>     ivadomed 1.2 depends on torchvision>=0.6.0
>     ivadomed 1.1 depends on torchvision>=0.6.0
>     ivadomed 1.0 depends on torchvision>=0.6.0
>     ivadomed 0.14 depends on torchvision>=0.6.0
>     ivadomed 0.13 depends on torchvision>=0.6.0
>     ivadomed 0.12 depends on torchvision>=0.6.0
>     ivadomed 0.11 depends on torchvision>=0.6.0
> 
> To fix this you could try to:
> 1. loosen the range of package versions you've specified
> 2. remove package versions to allow pip attempt to solve the dependency conflict
> 
> ERROR: ResolutionImpossible: for help visit https://pip.pypa.io/en/latest/user_guide/#fixing-conflicting-dependencies
> ```
> 
> Specifically, the error seems to be with torchvision:
> 
> ```
> Collecting ivadomed==2.6.1
>   Using cached ivadomed-2.6.1-py3-none-any.whl (145 kB)
> ERROR: Could not find a version that satisfies the requirement torchvision==0.6.0 (from ivadomed)
> ERROR: No matching distribution found for torchvision==0.6.0
> ```
> 
> I tried this with various python versions to no avail. Overall, this seems to be related to WindowsOS, as the installation proceeds fine on other OS. I would suggest updating your RTD installation section and the README to state installation works as described for MacOS and Linux, but not currently supported for Windows. It may be something you want to fix or not - but this will clarify upfront to users that installation works as intended except on Windows and avoid head scratching at the first stage.
> 
> # Statement of Need
> 
> There is a table in the paper linking to a few of the packages in the scope of medical imaging and deep learning - as you note the field rapidly evolves and keeping track of all the packages is tough, so this table is useful. However, could I suggest to change the last sentence of the paper to highlight the completeness of your package in comparison to the others?
> 
> In your table you include some examples of registration packages that use deep learning, yet ivadomed does not yet cover these functionalities (yet (?)). Are you planning on covering this use case in the future? Additionally, would you consider explaining this in your paper, and adding [DeepReg](https://doi.org/10.21105/joss.02705) to your table/citing it?

@NMontanaBrown  Can you verify the version of Python used on each of these operating systems?

**Comment by NMontanaBrown** [comment]
@ssinad 

I used conda for installation. So, on all OS, do:

`conda create --name ivadomed python=3.x`
`pip install ivadomed`

so ivadomed latest: 2.6.1

- MacOS: 3.7.1, 3.6, 3.8. 3.9:  downloads all the wheels for ivadomed all down to 1.1, and installs the 1.1 package.
- Ubuntu 18.04.3: 3.6, 3.7.9, 3.8.5, 3.9: downloads the wheels as with macOS - installs ivadomed 1.1.
- Windows OS: None, from 3.6 - 3.9.

**Comment by ssinad** [comment]
@NMontanaBrown 
Can you try this one instead?
```
pip install ivadomed -f https://download.pytorch.org/whl/torch_stable.html
```

**Comment by NMontanaBrown** [comment]
> @NMontanaBrown
> Can you try this one instead?
> 
> ```
> pip install ivadomed -f https://download.pytorch.org/whl/torch_stable.html
> ```

With which OS?

**Comment by ssinad** [comment]
> > @NMontanaBrown
> > Can you try this one instead?
> > ```
> > pip install ivadomed -f https://download.pytorch.org/whl/torch_stable.html
> > ```
> 
> With which OS?

@NMontanaBrown
Preferably Windows, but feel free to test it with others too.

**Comment by charleygros** [comment]
> @NMontanaBrown
> Preferably Windows, but feel free to test it with others too.

We should move and fix this issue internally @ssinad! Thanks again for your valuable help @NMontanaBrown :-)

**Comment by NMontanaBrown** [comment]
> > > @NMontanaBrown
> > > Can you try this one instead?
> > > ```
> > > pip install ivadomed -f https://download.pytorch.org/whl/torch_stable.html
> > > ```
> > 
> > 
> > With which OS?
> 
> @NMontanaBrown
> Preferably Windows, but feel free to test it with others too.

Apologies, this slipped my mind.

I have tried this with Windows, it seems to fix the problem, install proceeds correctly on Windows! 🍾 😄 

I have not tried it with Linux or MacOS as these install fine as is.

**Comment by charleygros** [comment]
Many thanks again @NMontanaBrown :-)

**Comment by charleygros** [comment]
@cMadan: We are ready for the next step I think :-)

**Comment by cMadan** [comment]
@charleygros, apologies for the delay, things have been quite busy. I'll do my final checks now and then let you know what are the last steps here.

**Comment by cMadan** [comment]
@whedon check references

**Comment by whedon** [comment]
```
Reference check summary (note 'MISSING' DOIs are suggestions that need verification):

OK DOIs

- 10.1109/iccv.2019.00707 is OK
- 10.1016/j.nicl.2018.03.029 is OK
- 10.1016/j.neuroimage.2018.09.081 is OK
- 10.14245/ns.1938396.198 is OK
- 10.1038/sdata.2016.44 is OK
- 10.1016/j.media.2019.101557 is OK
- 10.1016/j.neucom.2019.01.103 is OK
- 10.1007/978-3-319-46723-8_54 is OK
- 10.1109/iccvw.2017.9 is OK
- 10.1109/CVPR.2016.90 is OK
- 10.1016/j.neuroimage.2018.11.042 is OK
- 10.1109/3dv.2016.79 is OK
- 10.1007/978-3-030-00931-1_70 is OK

MISSING DOIs

- None

INVALID DOIs

- None
```

**Comment by cMadan** [comment]
@whedon generate pdf

**Comment by whedon** [comment]
:point_right::page_facing_up: [Download article proof](https://raw.githubusercontent.com/openjournals/joss-papers/joss.02868/joss.02868/10.21105.joss.02868.pdf) :page_facing_up: [View article proof on GitHub](https://github.com/openjournals/joss-papers/blob/joss.02868/joss.02868/10.21105.joss.02868.pdf) :page_facing_up: :point_left:

**Comment by cMadan** [comment]
@NMontanaBrown @justusschock @lbugnon, thank you for the thorough reviews, they are much appreciated!

@charleygros, everything looks good to me!

To move forward with accepting your submission, there are a few last things to take care of:

- [ ] Make a tagged release of your software, and list the version tag of the archived version here. 
- [ ] Archive the reviewed software in Zenodo 
- [ ] Check the Zenodo deposit has the correct metadata, this includes the title (should match the paper title) and author list (make sure the list is correct and people who only made a small fix are not on it); you may also add the authors' ORCID.
- [ ] List the Zenodo DOI of the archived version here.

You may find this helpful: https://guides.github.com/activities/citable-code/

**Comment by charleygros** [comment]
Many thanks for your help @cMadan !

Release of the software done --> [v2.7.0](https://github.com/ivadomed/ivadomed/tree/v2.7.0)

**Comment by charleygros** [comment]
And here is the Zenodo DOI @cMadan:
```
10.5281/zenodo.4527802
```

**Comment by cMadan** [comment]
@whedon set v2.7.1 as version

**Comment by whedon** [comment]
OK. v2.7.1 is the version.

**Comment by cMadan** [comment]
@whedon set 10.5281/zenodo.4527802 as archive

**Comment by whedon** [comment]
OK. <a href="https://doi.org/10.5281/zenodo.4527802" target="_blank">10.5281/zenodo.4527802</a> is the archive.

**Comment by cMadan** [comment]
@whedon accept

**Comment by whedon** [comment]
```
Attempting dry run of processing paper acceptance...
```

**Comment by whedon** [comment]
:wave: @openjournals/joss-eics, this paper is ready to be accepted and published.

 Check final proof :point_right: https://github.com/openjournals/joss-papers/pull/2084

If the paper PDF and Crossref deposit XML look good in https://github.com/openjournals/joss-papers/pull/2084, then you can now move forward with accepting the submission by compiling again with the flag `deposit=true` e.g.
 ```
@whedon accept deposit=true
```

**Comment by whedon** [comment]
```
Reference check summary (note 'MISSING' DOIs are suggestions that need verification):

OK DOIs

- 10.1109/iccv.2019.00707 is OK
- 10.1016/j.nicl.2018.03.029 is OK
- 10.1016/j.neuroimage.2018.09.081 is OK
- 10.14245/ns.1938396.198 is OK
- 10.1038/sdata.2016.44 is OK
- 10.1016/j.media.2019.101557 is OK
- 10.1016/j.neucom.2019.01.103 is OK
- 10.1007/978-3-319-46723-8_54 is OK
- 10.1109/iccvw.2017.9 is OK
- 10.1109/CVPR.2016.90 is OK
- 10.1016/j.neuroimage.2018.11.042 is OK
- 10.1109/3dv.2016.79 is OK
- 10.1007/978-3-030-00931-1_70 is OK

MISSING DOIs

- None

INVALID DOIs

- None
```

**Comment by jcohenadad** [comment]
i noticed a broken link in the generated pdf. Page 1, the hyperlink under "each has their pros & cons." brings me there:
![image](https://user-images.githubusercontent.com/2482071/107651948-0dd14700-6c4e-11eb-843f-e98941d6de8f.png)

**Comment by jcohenadad** [comment]
Same issue for hyperlink under "this example tutorial" in page 3:

![image](https://user-images.githubusercontent.com/2482071/107652375-82a48100-6c4e-11eb-82d8-facc42529c20.png)

The issue is possibly related to the "#" character in the URL (interpreted as %23)?

**Comment by jcohenadad** [comment]
about the broken links mentioned above: when opening the generated PDF with acrobat reader, the links work. With macOs preview, i get those errors.

**Comment by danielskatz** [comment]
For what it's worth, they work for me in Safari on macOS

**Comment by jcohenadad** [comment]
@cMadan have you had similar issues with generated PDFs in the past? What do you suggest we can do? (e.g., remove the "#" in the URLs, but it would be a less direct redirection 🤷‍♂️)

**Comment by jcohenadad** [comment]
> For what it's worth, they work for me in Safari on macOS

@danielskatz It doesn't work for me on Safari (see below). I'm running macOs 10.15.7 with Safari 14.0 and opened the PDF with Safari from my local station. I could not open it from the link in the [GitHub PR](https://github.com/openjournals/joss-papers/pull/2084/files?file-filters%5B%5D=.pdf#diff-c8d0a82b4ebde1bcd2aceb93d3cd0d9627c7b695af3899135d6f5cac8323fb4f).

![Screen Shot 2021-02-11 at 11 21 14 AM](https://user-images.githubusercontent.com/2482071/107665619-9e625400-6c5b-11eb-9cb5-d41608d5ef71.png)

**Comment by danielskatz** [comment]
I'm on macOS 11.2 and Safari 14.0.3

**Comment by justusschock** [comment]
Works for me on macOS 11.2 with both, safari and chrome

**Comment by jcohenadad** [comment]
Just for reference, this is a known issue. In [this thread](https://github.com/latex3/hyperref/issues/110), they suggest to replace the hyperlinks with bitly links, but I don't know if we could rely on them in the long term.

**Comment by danielskatz** [comment]
Given that this is resolved on current macOS, this appears to be a bug apple fixed, so I don't think a workaround is needed as we go forward.

**Comment by Drulex** [comment]
In the genreated link, the fragment uri (ascii character `#`) is url-encoded (`%23`) which seems appropriate, but apparently not handled the same way by all browsers.

Maybe you can try tweaking whatever process generated the hyperlink to use `ascii` instead of url-encoded links? (warning this may break stuff, especially with hyperlinks that have embedded spaces)

**Comment by cMadan** [comment]
fwiw. The links work correctly in Windows Chrome as well.

**Comment by arfon** [comment]
@whedon accept deposit=true

**Comment by whedon** [comment]
```
Doing it live! Attempting automated processing of paper acceptance...
```

**Comment by whedon** [comment]
🐦🐦🐦 👉 [Tweet for this paper](https://twitter.com/JOSS_TheOJ/status/1360241610139836422) 👈 🐦🐦🐦

**Comment by whedon** [comment]
🚨🚨🚨 **THIS IS NOT A DRILL, YOU HAVE JUST ACCEPTED A PAPER INTO JOSS!** 🚨🚨🚨

 Here's what you must now do:

0. Check final PDF and Crossref metadata that was deposited :point_right: https://github.com/openjournals/joss-papers/pull/2085
1. Wait a couple of minutes to verify that the paper DOI resolves [https://doi.org/10.21105/joss.02868](https://doi.org/10.21105/joss.02868)
2. If everything looks good, then close this review issue.
3. Party like you just published a paper! 🎉🌈🦄💃👻🤘

 Any issues? Notify your editorial technical team...

**Comment by arfon** [comment]
@NMontanaBrown, @justusschock, @lbugnon - many thanks for your reviews here and to @cMadan for editing this submission ✨. JOSS relies upon the volunteer efforts of folks like yourselves and we simply wouldn't be able to do this without you!

@charleygros - your paper is now accepted and published in JOSS :zap::rocket::boom:

**Comment by whedon** [comment]
:tada::tada::tada: Congratulations on your paper acceptance! :tada::tada::tada:

If you would like to include a link to your paper from your README use the following code snippets:

```
Markdown:
[![DOI](https://joss.theoj.org/papers/10.21105/joss.02868/status.svg)](https://doi.org/10.21105/joss.02868)

HTML:
<a style="border-width:0" href="https://doi.org/10.21105/joss.02868">
  <img src="https://joss.theoj.org/papers/10.21105/joss.02868/status.svg" alt="DOI badge" >
</a>

reStructuredText:
.. image:: https://joss.theoj.org/papers/10.21105/joss.02868/status.svg
   :target: https://doi.org/10.21105/joss.02868
```
This is how it will look in your documentation:

[![DOI](https://joss.theoj.org/papers/10.21105/joss.02868/status.svg)](https://doi.org/10.21105/joss.02868)

**We need your help!**

Journal of Open Source Software is a community-run journal and relies upon volunteer effort. If you'd like to support us please consider doing either one (or both) of the the following:

- Volunteering to review for us sometime in the future. You can add your name to the reviewer list here: https://joss.theoj.org/reviewer-signup.html
- Making a small donation to support our running costs here: https://numfocus.org/donate-to-joss

**Comment by jcohenadad** [comment]
@NMontanaBrown, @justusschock, @lbugnon: many thanks for your reviews, @cMadan for editing the submission and @arfon for helping with everything. We had an awesome experience publishing with JOSS and we will certainly recommend it around! Keep up the good work! 🙏

**Comment by charleygros** [comment]
@cMadan: Any idea how we can help Google Scholar to reference the JOSS paper? They found the arxiv version but not the JOSS one. Cheers

**Comment by cMadan** [comment]
There's nothing for you to specifically do, but this is something we've been working on over the years (see https://github.com/openjournals/joss/issues/130 for some context)

**Comment by arfon** [comment]
@charleygros - as far as I am aware, Scholar _does_ index JOSS these days, but it's possible this happens less frequently than the arXiv?
